File: Applicationjava/myviewpager/src/main/java/com/ly/myviewpager/MainActivity3.java
package com.ly.myviewpager;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;
import com.google.android.material.tabs.TabLayout;

import java.util.ArrayList;
import java.util.List;

public class MainActivity3 extends AppCompatActivity {
    private ViewPager vp;
    private TabLayout tl;

    List<Fragment> fragments = new ArrayList<>();
    List<String> titles = new ArrayList<>();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main3);
        vp = findViewById(R.id.vp);
        tl = findViewById(R.id.tl);
        fragments.add(AFragment.newInstance("11111", "11111"));
        fragments.add(BFragment.newInstance("22222", "22222"));
        fragments.add(CFragment.newInstance("33333", "33333"));
        titles.add("fragment1");
        titles.add("fragment2");
        titles.add("fragment3");
        tl.setupWithViewPager(vp);
        vp.setAdapter(new FragmentPagerAdapter(getSupportFragmentManager(),FragmentPagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
            @NonNull
            @Override
            public Fragment getItem(int position) {
                return fragments.get(position);
            }

            @Override
            public int getCount() {
                return fragments.size();
            }

            @Nullable
            @Override
            public CharSequence getPageTitle(int position) {
                return titles.get(position);
            }
        });


    }
}

File: Applicationjava/mylifecycle/src/main/java/com/ly/mylifecycle/MainActivity3.kt
package com.ly.mylifecycle

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity3 : AppCompatActivity() {
    private lateinit var myLifecycleOwner: MyLifecycleOwner
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main3)
        //自定义lifecycle owner 添加监听器
        myLifecycleOwner = MyLifecycleOwner()
        myLifecycleOwner.lifecycle.addObserver(MyLifecycleObserver())
        //改变状态
        myLifecycleOwner.create()
    }

    override fun onStart() {
        super.onStart()
        myLifecycleOwner.start()
    }

    override fun onResume() {
        super.onResume()
        myLifecycleOwner.resume()
    }
}